### PR TITLE
feat: welcome message and string functions edits

### PR DESCRIPTION
**Pull request by EttyKitty**
### The welcome message:
- Fixed wrong names due to wrong array indices.
- Added units that were missed from the listing: Champions, Librarians, Sergeants, Veteran Sergeants.
- Edited bits of text a little and changed some formatting, so it's more nice and readable (I hope).
- Increased the amount of info on each page after the first, removing the need for the 4th page.
- Fixed equipment list starting at index 1, causing it to skip items at 0 index (bolt pistols, most of the time).
- Various fixes to words using/not using plural forms when required.

### Code:
- Various minor refactors, style edits, etc.
- Replaced a lot of string concentration with interpolation.
- Extremely prevalent use of `string_plural`.

### Functions:
- Moved all string functions into one file, "scr_string_functions", just like "scr_array_functions".
- Placed `handle_exception` into all (most?) catch. Should work like a light version of try_and_report.
- `string_plural` now supports a second argument - variable to check to decide whenever to convert a string to plural form or not.
- New function `integer_to_letters` - converts an integer to a string word. Supports 2 additional arguments - capitalize the first letter of the resulting word; output an ordinal form of the word.

## Notes:
- Originally tried using int to word function, but decided that while more immersive, it's a bit less readable. Still not sure about it.

## TODO:
- [x] Fill documentation on all string functions.
- [x] Rename `integer_to_letters` into `integer_to_word`.
- [x] Fix total page count showing 4 pages, even though there are 3.

### CODE REVIEW

File: objects/obj_controller/Draw_0.gml [hunk 1]
```diff
@@ -3,29 +3,29 @@
 try{
     scr_ui_manage();
 } catch(_exception){
```
Comment (OH296) [inline objects/obj_controller/Draw_0.gml:5]: in general while messing around would be best to sling these in `try_and_report_loop` maybe?

**Review by OH296 — APPROVED** [review]
made one point but probably outside of scope for pr

File: objects/obj_controller/Draw_0.gml (hunk 1 above)
Comment (EttyKitty) [reply objects/obj_controller/Draw_0.gml:5]: > in general while messing around would be best to sling these in `try_and_report_loop` maybe?

I think try/catch and `handle_exception` combo should do the same thing.
I've created `handle_exception` as a light version of `try_and_report_loop`, for cases where you don't need custom functions to run on catch. It's just more compact, but works mostly the same way, I think?

File: objects/obj_controller/Create_0.gml [hunk 2]
```diff
@@ -1497,33 +1499,34 @@ for(var mm=1; mm<=100; mm++){
...
-if (libra>0) then temp[njm]+=", "+string(libra)+" "+string(obj_ini.role[100,17])+"s";
-if (coda>0) then temp[njm]+=", "+string(coda)+" Codiciery";
-if (lexa>0) then temp[njm]+=", "+string(lexa)+" Lexicanum.";
+temp[njm] = "Specialist branches staffed by";
+if (techa > 0) then temp[njm] += $", {techa} {string_plural(obj_ini.role[100][16], techa)}";
+if (old_dudes > 0) then temp[njm] += $", {old_dudes} {string_plural(obj_ini.role[100][14], old_dudes)}";
+if (apotha > 0) then temp[njm] += $", {apotha} {string_plural(obj_ini.role[100][15], apotha)}";
+if (libra > 0) then temp[njm] += $", {libra} {string_plural(obj_ini.role[100,17], libra)}";
+if (coda > 0) then temp[njm] += $", {coda} {string_plural("Codiciery")}";
+if (lexa > 0) then temp[njm] += $", {lexa} {string_plural("Lexicanum")}.";
```
Comment (EttyKitty) [inline objects/obj_controller/Create_0.gml]: Add a second argument to check for the count.

File: objects/obj_controller/Draw_0.gml (hunk 1 above)
Comment (OH296) [reply objects/obj_controller/Draw_0.gml:5]: i guess that's true, i suppose i was mostly just thinking that ```try_and_report_loop("ui Manage"), scr_ui_manage)``` is just less lines and coudl have as you say a custom catch function, but it's probably outside of scope for this pr

**Review by OH296 — APPROVED** [review]

**Pull request merged**